azurelinuxagent/common/protocol/wire.py

```diff
@@ -692,6 +715,16 @@ def update_certs(self, goal_state):
         self.save_cache(local_file, xml_text)
         self.certs = Certificates(self, xml_text)
 
+    def update_remote_access_conf(self, goal_state):
+        if goal_state.remote_access_uri is not None:            
+            xml_text = self.fetch_config(goal_state.remote_access_uri, self.get_header_for_cert())
+        else:
+            logger.verbose("{0}: remote access uri is empty.".format(sys._getframe().f_code.co_name))
+            xml_text = ""
+        local_file = os.path.join(conf.get_lib_dir(), REMOTE_ACCESS_FILE_NAME)
```

Comment:
Is this for each incarnation or is really a global value?  If it is per incarnation, please include the incarnation number in the file name.

Reply:
Right now it's a global value as we just re-write the file.  I can't think of a reason to keep previous versions of the file around.